src/stdio_mgr/stdio_mgr.py

```diff
@@ -190,6 +165,45 @@ def readline(self, size=-1):
         self.tee.write(text)
         return text
 
+
+class SimulateStdin(TextIOWrapper):
+    """Class to simulate content appearing on stdin.
+
+    Subclass of :class:`~io.TextIOWrapper` that
+    provides :meth:`~SimulateStdin.getvalue` which emulates the
+    behavior of :meth:`StringIO.getvalue() <io.StringIO.getvalue>`,
+    decoding the buffer using the :attr:`~io.TextIOBase.encoding`.
+
+    This class also provides the method
+    :meth:`~SimulateStdin.append`, which is not available
+    for the base :class:`TextIOWrapper <io.TextIOBase>` type.
+    This method adds new content to the end of the
+    stream while leaving the read position unchanged.
+
+    As a subclass of :class:`~io.TextIOWrapper`, it is not thread-safe.
+
+    Instantiation takes two arguments:
+
+    `init_text`
+
+        |str| *(optional)* --
+        Text to use as the initial contents of the
+        underlying :class:`~io.TextIOWrapper`. `init_text` is
+        passed directly to the :class:~io.TextIOWrapper`
+        instantiation call. Default is an empty |str|.
```

Comment:
Two issues
- "underlying TextIOWrapper" isnt correct because it is a subclass of TextIOWrapper, and "underlying" is normally used to describe containment rather than inheritance.
- "passed directly" is not correct as you mentioned.
Will edit.